src/binaries/local/main.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_base::base::tokio;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_config::Config;
+use common_meta_embedded::MetaEmbedded;
+use common_tracing::set_panic_hook;
+use databend_query::GlobalServices;
+use databend_query::servers::http::v1::{ExecuteStateKind, HttpQueryContext};
+use databend_query::servers::http::v1::query::{HttpQuery, HttpQueryConfig, HttpQueryRequest};
+use databend_query::sessions::{SessionManager, SessionType};
+
+use clap::Parser;
+use serde::Deserialize;
+use serde::Serialize;
+use prettytable::{Table, Row, Cell};
+use databend_query::servers::http::v1::json_block::JsonBlock;
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Parser)]
+#[clap(about, author)]
+pub struct LocalConfig {
+    // sql to run
+    #[clap(long, default_value = "SELECT 1")]
+    pub sql: String,
+
+    // name1=filepath1,name2=filepath2
+    #[clap(long, default_value = "")]
+    pub table: String,
+
+    #[clap(long, default_value = "INFO")]
+    pub log_level: String,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), ErrorCode> {
+    main_entrypoint().await?;
+    Ok(())
+}
+
+async fn main_entrypoint() -> Result<()> {
+    let local_conf = LocalConfig::parse();
+    let conf: Config = Config::load_for_test()?;
+
+    set_panic_hook();
+    MetaEmbedded::init_global_meta_store(conf.meta.embedded_dir.clone()).await?;
+
+    // Make sure global services have been inited.
+    GlobalServices::init(conf.clone()).await?;
+
+    let session = SessionManager::instance().create_session(SessionType::Local).await?;
+    let httpctx = HttpQueryContext::new(session.clone());
```

Comment:
Yes, you should query directly through:

```
  let mut planner = Planner::new(context.clone());
  let (plan, _, fmt) = planner.plan_sql(&sql).await.map_err(BadRequest)?;
  let format = get_format_with_default(fmt, default_format)?;

  context.attach_query_str(plan.to_string(), &sql);
  let interpreter = InterpreterFactory::get(context.clone(), &plan)
      .await
      .map_err(BadRequest)?;
  execute(context, interpreter, plan.schema(), format, params, None)

```

Maybe you don't need to pretty the block by yourself, you can pass the format into SQL, eg:

```
select * from tb1 format NDJSON
```

Reply:
Should we reuse the `clickhouse_handler` or write a new `local_handler`?

Reply:
I do not think we need any handlers. Just follow:

` SQL --> Parser --> Plan --> Interpreter --> execute --> blocks`

Reply:
Fixed, PTAL~